packages/jest-mock/src/index.ts

```diff
@@ -20,87 +20,101 @@ export type MockFunctionMetadataType =
 export type MockFunctionMetadata<
   T,
   Y extends Array<unknown>,
-  Type = MockFunctionMetadataType,
+  MetadataType = MockFunctionMetadataType,
 > = {
   ref?: number;
   members?: Record<string, MockFunctionMetadata<T, Y>>;
   mockImpl?: (...args: Y) => T;
   name?: string;
   refID?: number;
-  type?: Type;
+  type?: MetadataType;
   value?: T;
   length?: number;
 };
 
-export type MockableFunction = (...args: Array<any>) => any;
-export type MethodKeysOf<T> = {
-  [K in keyof T]: T[K] extends MockableFunction ? K : never;
+export type ConstructorLike = {new (...args: Array<any>): any};
+
+export type MethodLike = (...args: Array<any>) => any;
```

Comment:
can we use `CallableFunction` from ts?

Reply:
Interesting idea. I was checking if `CallableFunction` (also `NewableFunction`) could work here, but it didn’t. Seems like these are great types for `bind`, `apply`, but what we need here is just to check that type has callable signature. That’s why adding `Like` looked good, but it also might be good idea to name them simply: `Callable` and `Constructable`. What you think?